assets/js/src/tracker/index.js

```diff
@@ -0,0 +1,104 @@
+import { config } from '../config';
+
+/**
+ * A tracking utility for initializing a GA4, managing accepted
+ * events, attaching data to those events, and tracking them.
+ *
+ * @param {Object} config Configuration settings for the tracker.
+ *
+ * @return {Object} An object containing methods: init, setupEvents, event{ attach, get, track }.
+ */
+export const tracker = ( () => {
+	const events = [];
+
+	return {
+		/**
+		 * Initializes the tracker and dataLayer if not already done.
+		 */
+		init: () => {
```

Comment:
:nail_care: 
Instead of object literal and `init`, I'd use class syntax to create a singleton, something like
```js
class Tracker{
	/**
	 * Initializes the tracker and dataLayer if not already done.
	 */
	constructor() {
	  // We can protect against `new tracker.constructor()` if we want to
	  if(instance){
	  	throw new Error("New instance cannot be created!!");
	  	// or more gently, but a bit tricky
	        return instance;
	    }
	    instance = this;
//...

export const tracker = Object.freeze(new Tracker());
```

Then we can add regular methods and properties. Allowing us to reason and inspect the object more structurally.